File: penghitung bmi.py
def ulang():

    berat = int(input("Masukkan berat badan anda (kg) : "))
    tinggiCM = int(input("Masukkan tinggi badan anda (cm) : "))
    tinggiM = tinggiCM*0.01
    bmi = berat/(tinggiM*tinggiM)
    idealL = (tinggiCM - 100)-(tinggiCM - 100)*0.1
    idealP = (tinggiCM - 100)-(tinggiCM - 100)*0.15
    bmiR = round(bmi, 2)
    print("BMI anda adalah : "+str(bmiR))
    if bmiR > 23:
        print("Anda kelebihan berat badan, berat badan yang ideal untuk anda adalah : ")
        print("Laki-laki : "+str(idealL)+" kg")
        print("Perempuan : "+str(idealP)+" kg")
    elif bmiR < 18.5:
        print("Anda kekurangan berat badan, berat badan yang ideal untuk anda adalah : ")
        print("Laki-laki : "+str(idealL)+" kg")
        print("Perempuan : "+str(idealP)+" kg")
    else:
        print("Berat badan anda normal")
    pengulangan = input("Lakukan perhitungan lagi? ya / tidak : ")
    
    if pengulangan == "ya":
        ulang()
    elif pengulangan == "tidak":
        print("Ok")
    else:
        print("Saya tidak mengerti perintah anda, saya akan menganggap anda menjawab tidak.")
ulang()

File: penghitung bmi.lua
function ulang()
  
  io.write("Masukkan berat badan anda (kg) : ")
  berat = tonumber(io.read())
  io.write("Masukkan tinggi badan anda (cm) : ")
  tinggiCM = tonumber(io.read())
  tinggiM = tinggiCM*0.01
  bmi = berat/(tinggiM*tinggiM)
  idealL = (tinggiCM - 100)-(tinggiCM - 100)*0.1
  idealP = (tinggiCM - 100)-(tinggiCM - 100)*0.15
  
  print("BMI anda adalah : "..tostring(bmi))
  
  if bmi > 23 then
    print("Anda kelebihan berat badan, berat badan yang ideal untuk anda adalah : ")
    print("Laki-laki : "..tostring(idealL).." kg")
    print("Perempuan : "..tostring(idealP).." kg")
  elseif bmi < 18.5 then
    print("Anda kekurangan berat badan, berat badan yang ideal untuk anda adalah : ")
    print("Laki-laki : "..tostring(idealL).." kg")
    print("Perempuan : "..tostring(idealP).." kg")
  else
     print("Berat badan anda normal")
  end
  
  io.write("Lakukan perhitungan lagi? ya / tidak : ")
  pengulangan = io.read()
      
  if pengulangan == "ya" then
    ulang()
  elseif pengulangan == "tidak" then
    print("Ok")
  else
    print("Saya tidak mengerti perintah anda, saya akan menganggap anda menjawab tidak.")
  end
end
ulang()
